test/modules/layers/test_attention.py

```diff
@@ -17,164 +19,300 @@
 )
 
 
-class TestAttention(unittest.TestCase):
-    """
-    Test all Attention classes
-    """
+@pytest.fixture(autouse=True)
+def set_seed():
+    set_rng_seed(4)
+
+
+@pytest.fixture
+def hidden_dim():
+    return 3
+
+
+@pytest.fixture
+def n_heads():
+    return 1
+
+
+@pytest.fixture
+def n_dim():
+    return 3
+
+
+@pytest.fixture
+def input_shape(n_dim):
+    return tuple(repeat(2, n_dim))
+
+
+@pytest.fixture
+def q(n_heads, input_shape, hidden_dim):
+    return torch.randn(1, n_heads, *input_shape, hidden_dim // n_heads)
+
+
+@pytest.fixture
+def kv(n_heads, input_shape, hidden_dim):
+    return torch.randn(1, n_heads, *input_shape, hidden_dim // n_heads)
+
+
+@pytest.fixture
+def full_attn(input_shape):
+    # TODO: retire causal once mask generation is moved out of FullAttention
+    #   causal inside FullAttention does not affect caching of k, v
```

Comment:
nit: move this TODO comment outside of test and in FullAttention for easier tracking?